### PR TITLE
[IMP] knowledge: improve wording and design of the editor view

### PR DESCRIPTION
**Pull request by julien-banken**
- Improve the wording of the confirmation modals to give clearer indication on what would happen when if the user confirms the action.
- Update the placeholder of the recipients field.
- Improve alignment of the "Access Restricted" warning of the share panel.
- Improve the design of the invite button of the share panel.

task-2893088
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/knowledge/static/src/js/widgets/knowledge_dialogs.js [hunk 1]
```diff
@@ -177,6 +177,40 @@ const MoveArticleToDialog = Dialog.extend({
     },
 });
 
+const ConfirmMoveDialog = Dialog.extend({
```
Comment (dbeguin) [inline addons/knowledge/static/src/js/widgets/knowledge_dialogs.js:180]: If I understood well the goal of this dialog (set Private or Workspace in bold) this component does not worth it. We should keep what have been done and live with it.
And Luc is maybe sad but agrees this is not a must have (and we can live without a bold category in the confirm message)

File: addons/knowledge/views/knowledge_article_views.xml [hunk 2]
```diff
@@ -84,12 +84,11 @@
...
-                                                    <button type="action" name="%(knowledge_invite_action_from_article)d"
-                                                            class="btn btn-primary w-100"
-                                                            attrs="{'invisible': [('user_has_write_access', '=', False)]}">
-                                                        <i class="fa fa-plus"/> Invite
-                                                    </button>
+                                                <div class="o_knowledge_share_panel_invite input-group o_cursor_pointer px-3 pb-2" attrs="{'invisible': [('user_has_write_access', '=', False)]}">
+                                                    <div class="form-control">Add people or email addresses</div>
+                                                    <div class="input-group-append">
+                                                        <button class="btn btn-primary">Invite</button>
+                                                    </div>
```
Comment (dbeguin) [inline addons/knowledge/views/knowledge_article_views.xml:91]: could be done without any added js by doing : 
```
<button type="action" name="%(knowledge_invite_action_from_article)d"
             class="input-group o_cursor_pointer px-3 pb-2"
             attrs="{'invisible': [('user_has_write_access', '=', False)]}">
    <div class="form-control text-left">Add people or email addresses...</div>
    <div class="input-group-append">
         <div class="btn btn-primary">Invite</div>
    </div>
 </button>
```

**Review by dbeguin — COMMENTED** [review]
Hello Julien!
I have some remarks. Can mostly be simplified :)

File: addons/knowledge/views/knowledge_article_views.xml (hunk 2 above)
Comment (julien-banken) [reply addons/knowledge/views/knowledge_article_views.xml]: It does not work with Bootstrap 5 :cry: To call the action without listener, we need to use the button tag and this tag introduces many css rules that we would have to be overwrite to get the result we want. For the share panel rework, I will need to define that listener to intercept the click event and prevent the share panel from closing. So, I think it's okay to introduce it here.

File: addons/knowledge/static/src/js/widgets/knowledge_dialogs.js (hunk 1 above)
Comment (julien-banken) [reply addons/knowledge/static/src/js/widgets/knowledge_dialogs.js]: Ok, let's remove it.

**Pull request closed without merging**